### ISSUE TITLE
Hey You Pikachu Stuttering When VRU is plugged in

### ISSUE THREAD

**Issue by Ecksters** [raphnet/pj64raphnetraw#3]
Was working with the Project64 people on this issue, turns out it's a problem with this plugin, as I tried out Hey You, Pikachu! with NRage instead, and the lag I was having went away.
https://github.com/project64/project64/issues/1337

You can see the lag in [this video](https://www.youtube.com/watch?v=bLc04BPT8Ps).

The lag only occurs when the VRU is plugged in, if I use your plugin but only have 1 controller plugged in, there is no lag.

**Comment by raphnet** [comment]
But the VRU is not functional using the NRage plugin, correct?

I suppose this is due to the few milliseconds the back and forth through USB adds. Emulation is paused during the operation, and the game is likely not programmed to compensate for unexpected IO delays that do not occur on the real system.

But if I knew exactly how the VRU works, maybe I could think of something, but right now besides designing a new adapter working at USB High speed instead of Full speed, or discovering a way to reduce the round trip time, I don't see how I can fix this issue.

**Comment by Ecksters** [comment]
Correct, I just noted that even with your plugin, unplugging the VRU also resolves the issue.

I'm actually fairly confused as to why the VRU is communicating at all during the moments of extreme lag, essentially it appears to happen whenever the command is sent for the VRU to ignore input(when there's a red X on the talk icon).

While there might be some lag on level load since it's sending that level's accepted commands to the VRU, it seems like it may be possible to fix the lag whenever the VRU is told to ignore input.

**Comment by theboy181** [comment]
This PJ64 input driver kills performance for the emualtor in general, maybe this is the root of the issue?

**Comment by PFLI2** [comment]
I believe we're having the same issue. Seems 3 of the 5 issues open are related:
https://github.com/raphnet/pj64raphnetraw/issues/8
https://github.com/raphnet/pj64raphnetraw/issues/10
